unit_tests/utils/test_datetime_helpers.py

```diff
@@ -262,3 +264,160 @@ def test_epoch_millis():
     # Test roundtrip conversion
     dt3 = AirbyteDateTime.from_epoch_millis(dt.to_epoch_millis())
     assert dt3 == dt
+
+
+@pytest.mark.parametrize(
+    "input_value,expected_parser",
+    [
+        # Formats that use the whenever parser
+        ("2023-03-14", "whenever"),  # Date-only format
+        (1678806566, "whenever"),  # Unix timestamp
+        # Formats that use the dateutil parser
+        ("2023-03-14T15:09:26Z", "dateutil"),  # ISO format with T delimiter
+        ("2023-03-14T15:09:26+00:00", "dateutil"),  # ISO format with timezone
+        ("2023-03-14 15:09:26", "dateutil"),  # Missing T delimiter
+        ("14/03/2023 15:09:26", "dateutil"),  # Different date format
+    ],
+)
+def test_datetime_parser_selection(input_value, expected_parser, monkeypatch):
+    """Test that the correct parser is used based on the input format."""
+    # Create tracking variables
+    whenever_called = False
+    dateutil_called = False
+
+    # Store original functions
+    original_instant_module = __import__("whenever").Instant
+    original_parser_parse = parser.parse
+
+    # Create a spy for Instant.from_timestamp
+    def spy_from_timestamp(*args, **kwargs):
+        nonlocal whenever_called
+        whenever_called = True
+        # Call the original function
+        return original_instant_module.from_timestamp(*args, **kwargs)
+
+    # Create a spy for Instant.from_utc
+    def spy_from_utc(*args, **kwargs):
+        nonlocal whenever_called
+        whenever_called = True
+        # Call the original function
+        return original_instant_module.from_utc(*args, **kwargs)
+
+    # Create a spy for parser.parse
+    def spy_parser_parse(*args, **kwargs):
+        nonlocal dateutil_called
+        dateutil_called = True
+        # Call the original function
+        return original_parser_parse(*args, **kwargs)
+
+    # Create a mock Instant class with our spy methods
+    class MockInstant:
+        @staticmethod
+        def from_timestamp(*args, **kwargs):
+            return spy_from_timestamp(*args, **kwargs)
+
+        @staticmethod
+        def from_utc(*args, **kwargs):
+            return spy_from_utc(*args, **kwargs)
+
+        # Add any other methods that might be called
+        @staticmethod
+        def py_datetime():
+            return original_instant_module.py_datetime()
+
+    # Apply the mocks at the module level
+    monkeypatch.setattr("airbyte_cdk.utils.datetime_helpers.Instant", MockInstant)
+    monkeypatch.setattr("airbyte_cdk.utils.datetime_helpers.parser.parse", spy_parser_parse)
+
+    # Skip formats that would be rejected by validation checks
+    if isinstance(input_value, str) and "March" in input_value:
+        # Skip this test case as it would be rejected by validation
+        return
+
```

Comment:
Can we delete this?

```suggestion
    # Skip formats that would be rejected by validation checks
    if isinstance(input_value, str) and "March" in input_value:
        # Skip this test case as it would be rejected by validation
        return

```